lib/Word_Lib/Word_Lemmas_Internal.thy

```diff
@@ -416,4 +416,24 @@ lemma unat_minus_plus_one:
    using max_word_wrap apply blast
   by (metis less_is_non_zero_p1 word_less_nat_alt word_overflow_unat)
 
+lemma unat_max_word:
+  "2 ^ LENGTH('a) - 1 = unat (max_word :: 'a :: len word)"
```

Comment:
Wow, I would have thought we had a lemma like this already, but I can't find it either. Nice addition.